server/src/main/scala/io/delta/standalone/internal/DeltaSharedTableLoader.scala

```diff
@@ -287,8 +289,11 @@ class DeltaSharedTable(
     snapshot.version -> actions
   }
 
-  private def queryDataChangeSinceStartVersion(startingVersion: Long): Seq[model.SingleAction] = {
-    val latestVersion = tableVersion
+  private def queryDataChangeSinceStartVersion(
+      startingVersion: Long,
+      endingVersion: Option[Long]
+  ): Seq[model.SingleAction] = {
+    val latestVersion = tableVersion.min(endingVersion.getOrElse(tableVersion))
     if (startingVersion > latestVersion) {
       throw DeltaCDFErrors.startVersionAfterLatestVersion(startingVersion, latestVersion)
     }
```

Comment:
Check that endingVersion <= latestVersion?

Reply:
added.